.github/workflows/build.yml

```diff
@@ -1,17 +1,60 @@
-# Base: https://github.com/haskell/actions/tree/main/setup#usage
-
 name: build
 
 on:
   push:
     branches:
       - master
+    tags:
+      - "v*"
   pull_request:
 
 jobs:
+  prepare-release:
+    if: startsWith(github.ref, 'refs/tags/v')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone project
+        uses: actions/checkout@v2
+
+      - name: Create release
+        uses: softprops/action-gh-release@v1
+        with:
+          fail_on_unmatched_files: true
+          files: |
+            LICENSE
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
   build:
-    name: build
-    runs-on: ubuntu-latest # TODO matrix testing
+    name: build-${{ matrix.os }}
+    if: always()
+    needs: prepare-release
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-20.04
+            cache_path: ~/.stack
+            stack_args: "--test"
+            artifact_rel_path: /bin/dog-food
+            asset_name: dog-food-ubuntu-20_04-x86-64
+          - os: ubuntu-18.04
+            cache_path: ~/.stack
+            stack_args: "--test"
+            artifact_rel_path: /bin/dog-food
+            asset_name: dog-food-ubuntu-18_04-x86-64
+          # TODO re-enable tests for mac and windows when they consistently pass (remove stack_args altogether)
+          - os: macos-latest
+            cache_path: ~/.stack
+            stack_args: ""
+            artifact_rel_path: /bin/dog-food
+            asset_name: dog-food-macos-x86-64
+          - os: windows-latest
+            cache_path: C:/sr
+            stack_args: ""
+            artifact_rel_path: /bin/dog-food.exe
+            asset_name: dog-food-windows-x86-64
```

Comment:
add .exe?

Reply:
we'll have to rename it in install script anyway - adding .exe for windows seems similar to chmod +x for unix - maybe keep it the way it is?